api/handlers.go

```diff
@@ -0,0 +1,276 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/nhatvu148/business-day-go/models"
+	tools "github.com/nhatvu148/business-day-go/tools"
+	"github.com/rs/zerolog/log"
+)
+
+type BusinessDayResult struct {
+	Result bool   `json:"result"`
+	Error  string `json:"error"`
+}
+
+type CustomError struct {
+	msg string
+}
+
+func (m *CustomError) Error() string {
+	return m.msg
+}
+
+type CustomHoliday struct {
+	Id       int64  `json:"id"`
+	Date     string `json:"date"`
+	Category string `json:"category"`
+}
+
+type IdResponse struct {
+	Id int64 `json:"id"`
+}
+
+func (app *Application) CustomHolidayHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		query := r.URL.Query()
+		dateString := query.Get("date")
+		if dateString == "" {
+			// Get all Custom holidays
+			customHolidays, err := app.DB.GetCustomHolidays()
+
+			if err != nil {
+				log.Err(err).Msg("Get Custom holidays error")
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+
+			// Prevent Json Marshal from returning null when customHolidays is []
+			customHolidays1 := make([]models.CustomHoliday, 0)
+			if len(customHolidays) != 0 {
+				customHolidays1 = customHolidays
+			}
+
+			res, err := json.Marshal(customHolidays1)
+			if err != nil {
+				log.Err(err).Msg("JSON marshal error")
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+
+			w.Write(res)
+		} else {
+			isDateValid, date := tools.IsValidDate(dateString)
+
+			if !isDateValid {
+				log.Err(&CustomError{msg: "Invalid date"}).Msg("Invalid date")
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+
+			// Get Custom holiday by date
+			customHoliday, err := app.DB.GetCustomHolidayByDate(date)
+
+			if err != nil {
+				// How to get pq error code?
+				if err.Error() == "sql: no rows in result set" {
```

Comment:
Try to avoid testing for errors like this. Please follow the guidance here: https://go.dev/blog/go1.13-errors

If that doesn't make sense, I'd be happy to explain it.